addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3827,7 +3827,11 @@ options.registry.GridImage = options.Class.extend({
      * @returns {?HTMLElement}
      */
     _getImageGridItem() {
-        const parentEl = this.$target[0].parentNode;
+        let parentEl = this.$target[0].parentNode;
+        if (parentEl.matches("a")) {
+            // Case of an image link.
+            parentEl = parentEl.parentNode;
+        }
```

Comment:
Instead of

```javascript
let parentEl = this.$target[0].parentNode;
if (parentEl.matches("a")) {
    // Case of an image link.
    parentEl = parentEl.parentNode;
}
if (parentEl && parentEl.classList.contains('o_grid_item_image')) {
    return parentEl;
}
return null;
```

Can't it be:

```javascript
return this.$target[0].closest('.o_grid_item_image');
```

?

Reply:
I thought about it and preferred not doing it (it was safer to only look at the parent)
But now that I re-think about it, given that I removed the fact that only direct children are concerned in the CSS rules, it's better to change it indeed (after all, the class can only be added if the image are well like expected so there is no problem) :slightly_smiling_face: